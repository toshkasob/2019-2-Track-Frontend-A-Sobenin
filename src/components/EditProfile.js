import React from 'react';
import styles from '../styles/editProfileStyles.module.scss';

import { ReactComponent as EditProfileSvg } from '../images/camera_img.svg';

export default function EditProfile(props) {
	function handleSubmit(event) {
		event.preventDefault();
	}

	function handleInvalid(event) {
		event.target.setCustomValidity(
			'Should contain at least 5 and no more 32 letters or digits. (e.g. yo228)',
		);
	}

	return (
		<form
			id="edit_profile"
			className={styles.edit_profile}
			onSubmit={handleSubmit}
		>
			<div className={styles.profile_img_preview}>
				<EditProfileSvg className={styles.profile_img} />
			</div>
			<div
				className={`${styles.profile_input_div} ${styles.fullname_input_div}`}
			>
				<span className={styles.input_info}>Full name</span>
				<input className={styles.profile_input} data-qa="input-fullname" />
			</div>
			<div className={styles.comment2input_div} />
			<div
				className={`${styles.profile_input_div} ${styles.username_input_div}`}
			>
				<span className={styles.input_info}>Username</span>
				<input
					className={styles.profile_input}
					pattern="^[a-z][a-z0-9-_\.]{4,32}$"
					onInvalid={handleInvalid}
					data-qa="input-username"
				/>
			</div>
			<div className={styles.comment2input_div}>
				<span className={styles.comment2input}>
					At least 5 and no more 32 letters or digits
				</span>
			</div>
			<div
				className={`${styles.profile_input_div} ${styles.biodetails_input_div}`}
			>
				<span className={styles.input_info}>Bio</span>
				<textarea className={styles.profile_input} data-qa="input-bio" />
			</div>
			<div className={styles.comment2input_div}>
				<span className={styles.comment2input}>Any details about you</span>
			</div>
		</form>
	);
}
