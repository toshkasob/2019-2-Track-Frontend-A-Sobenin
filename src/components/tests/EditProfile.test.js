import React from 'react';
import renderer from 'react-test-renderer';

import EditProfile from '../EditProfile';

it('renders EditProfile correctly', () => {
	const profile = renderer
		.create(
			<div>
				<EditProfile />
			</div>,
		)
		.toJSON();
	expect(profile).toMatchSnapshot();
});
