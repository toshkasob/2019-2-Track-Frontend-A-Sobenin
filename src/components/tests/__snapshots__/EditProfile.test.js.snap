// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders EditProfile correctly 1`] = `
<div>
  <form
    className="edit_profile"
    id="edit_profile"
    onSubmit={[Function]}
  >
    <div
      className="profile_img_preview"
    >
      <svg
        className="profile_img"
      >
        camera_img.svg
      </svg>
    </div>
    <div
      className="profile_input_div fullname_input_div"
    >
      <span
        className="input_info"
      >
        Full name
      </span>
      <input
        className="profile_input"
      />
    </div>
    <div
      className="comment2input_div"
    />
    <div
      className="profile_input_div username_input_div"
    >
      <span
        className="input_info"
      >
        Username
      </span>
      <input
        className="profile_input"
        onInvalid={[Function]}
        pattern="^[a-z][a-z0-9-_\\\\.]{4,32}$"
      />
    </div>
    <div
      className="comment2input_div"
    >
      <span
        className="comment2input"
      >
        At least 5 and no more 32 letters or digits
      </span>
    </div>
    <div
      className="profile_input_div biodetails_input_div"
    >
      <span
        className="input_info"
      >
        Bio
      </span>
      <textarea
        className="profile_input"
      />
    </div>
    <div
      className="comment2input_div"
    >
      <span
        className="comment2input"
      >
        Any details about you
      </span>
    </div>
  </form>
</div>
`;
